nsis/src/nsis/tests/NsisExecutableTest.java

```diff
@@ -98,4 +97,23 @@ public void testNsisCreationBzipCompressed() throws IOException, InvalidFormatEx
 		}
 	}
 
+	/**
+	 * Deobfuscated the inputStream object passed as a parameter. The obfuscation is
+	 * a simple XOR with 0x55
+	 * 
+	 * @param inputStream
+	 * @return the deobfuscated ByteArrayProvider object
+	 * @throws IOException
+	 */
+	private ByteArrayProvider deobfuscate(InputStream inputStream) throws IOException {
+		byte[] obfuscated = inputStream.readAllBytes();
+		byte[] original = new byte[obfuscated.length];
+		int i = 0;
+		for (byte obfuscatedByte : obfuscated) {
+			original[i] = (byte) (obfuscatedByte ^ 0x55);
```

Comment:
nit: is a cast is required here?

Reply:
If I don't but the cast, it takes the `(obfuscatedByte ^ 0x55)` as an int and the linter is not happy and suggests adding a cast